src/test/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceTest.java

```diff
@@ -747,63 +754,78 @@ void shouldThrowInvalidRoleAssignmentExceptionWhenCaseTypeIsInvalidAndJurisdicti
-        when(caseHearingRequestRepository.getVersionNumber(2000000000L)).thenReturn(1);
-        when(hearingRepository.existsById(2000000000L)).thenReturn(true);
-        when(hearingRepository.getStatus(2000000000L)).thenReturn("UPDATE_SUBMITTED");
-        when(hearingRepository.findById(2000000000L)).thenReturn(Optional.of(TestingUtil.deleteHearingEntity()));
-        hearingManagementService.deleteHearingRequest(2000000000L, TestingUtil.deleteHearingRequest());
-        verify(hearingRepository).existsById(2000000000L);
-        verify(caseHearingRequestRepository).getVersionNumber(2000000000L);
-    }
-
-    @Test
-    void deleteHearingRequestShouldFailWithInValidStatus() {
-        when(caseHearingRequestRepository.getVersionNumber(2000000000L)).thenReturn(1);
-        when(hearingRepository.existsById(2000000000L)).thenReturn(true);
-        when(hearingRepository.getStatus(2000000000L)).thenReturn("UPDATE_NOT_SUBMITTED");
-        Exception exception = assertThrows(BadRequestException.class, () -> {
-            hearingManagementService.deleteHearingRequest(2000000000L, TestingUtil.deleteHearingRequest());
-        });
+        final long hearingId = 2000000000L;
+        HearingEntity hearingEntity = generateHearingEntity(hearingId, DeleteHearingStatus.UPDATE_REQUESTED.name(), 1);
+        when(caseHearingRequestRepository.getVersionNumber(hearingId)).thenReturn(1);
+        when(hearingRepository.existsById(hearingId)).thenReturn(true);
+        when(hearingRepository.getStatus(hearingId)).thenReturn("UPDATE_SUBMITTED");
+        when(hearingRepository.findById(hearingId)).thenReturn(Optional.of(hearingEntity));
+        when(hearingRepository.save(hearingEntity)).thenReturn(hearingEntity);
+        HearingResponse hearingResponse = hearingManagementService.deleteHearingRequest(
+                hearingId, TestingUtil.deleteHearingRequest());
+        assertEquals(CANCELLATION_REQUESTED, hearingResponse.getStatus());
+        verify(hearingRepository).existsById(hearingId);
+        verify(caseHearingRequestRepository).getVersionNumber(hearingId);
+    }
+
+    @Test
+    void deleteHearingRequestShouldFailWithInvalidStatus() {
+        final long hearingId = 2000000000L;
+        DeleteHearingRequest deleteHearingRequest = TestingUtil.deleteHearingRequest();
+        final int versionNumber = deleteHearingRequest.getVersionNumber();
+        when(caseHearingRequestRepository.getVersionNumber(hearingId)).thenReturn(versionNumber);
+        when(hearingRepository.existsById(hearingId)).thenReturn(true);
+        when(hearingRepository.getStatus(hearingId)).thenReturn("UPDATE_NOT_SUBMITTED");
+
+        Exception exception = assertThrows(BadRequestException.class, () ->
+            hearingManagementService.deleteHearingRequest(hearingId, deleteHearingRequest));
         assertEquals(INVALID_DELETE_HEARING_STATUS, exception.getMessage());
+        verify(hearingRepository).existsById(hearingId);
```

Comment:
can you add assert for version number pls

Reply:
Can't test version number as not available after Exception.